### ISSUE TITLE
Yocto doesn't pick up dependancies correct

### ISSUE THREAD

**Issue by htot** [edison-fw/meta-acpi#3]
@alext-mkrs Any ideas here?

When we change a sample aml file, bitbake doesn't feel like rebuilding the recipe. We need to do `bitbake -c cleansstate acpi-tables` first. And then, if we are unlucky, this doesn't rebuild the initramfs kernel. So need also `bitbake -c cleansstate linux-yocto`.

**Comment by alext-mkrs** [comment]
You mean [this](https://github.com/edison-fw/meta-acpi/blob/master/recipes-bsp/acpi-tables/acpi-tables.bb) recipe? If so, those tables are not really visible to bitbake, because we process them in the `do_compile()` function. Unless we introduce some custom class that would do something similar to what kernel config fragments one does, the simplest fix would be to tie the sample table list to the recipe, so that bitbake knows about list changes. I understand the original idea was flexibility, so that no list needs to be kept in the recipe, but OTMH I don't see any other solutions.

**Comment by alext-mkrs** [comment]
I'm not sure this is going to help - and that would require inheriting the automake class.

I'd suggest to just list them in `SRC_URI` variable. Then bitbake will know about them and will fetch them into the workdir, removing the need in `THISDIR`. At the expense of having the list in the recipe, we'll have the table change awareness.

**Comment by alext-mkrs** [comment]
I can BTW come up with the example of what I'm talking about - but again, either later this week or (most likely) on the weekend.

**Comment by htot** [comment]
No rush.
But as is, it is not very efficient. I need to `bitbake -c cleansstate acpi-tables`, followed by `bitbake acpi-tables`. And followed by `make image` (which cleans kernel, U-Boot). Otherwise the new table doesn't make it into the initramfs.
Or we could remove it from initramfs and load from U-Boot as intended, but then need to figure a way to add to the image so flashall will install it.

**Comment by htot** [comment]
I forgot about this: in `meta-intel-edison-bsp/conf/machine/edison.conf` we have 
`ACPI_TABLES ?= "arduino.asl spidev.asl"`

**Comment by alext-mkrs** [comment]
Yeah, but that's really just an internal variable we use in the recipe, it has no significance for bitbake. Let me see if I can come up with a working example of what I meant in my previous comment.

**Comment by htot** [comment]
I am probably missing the point. But the variable is defined outside the recipe, and could be used by the SRC_URI, so we remain flexible.

**Comment by alext-mkrs** [comment]
After further analysis I see this isn't going to be that easy. Given the current functionality and the intended goal of being able to pass ACPI tables not only from that `samples` dir, but also from arbitrary locations (by specifying full paths in `ACPI_TABLES`), using `SRC_URI` is not possible without constructing it during the recipe parsing stage or something like that. I've run out of time today and will need to get back to this laters, so if anyone has any other ideas - feel free to chime in.

**Comment by alext-mkrs** [comment]
@westeri, do you think this piece of functionality - passing arbitrary tables not only those in `samples` dir - is important enough to be preserved in these circumstances?

**Comment by westeri** [comment]
If it is not used anywhere, then just drop it ;-)
